### PR TITLE
Add Bootstrap Regression (Linear Regression)

### LINKED ISSUE #100: Bootstrap Regression for Linear Regression
**Bootstrap Regression**
Most of the time Normality assumption of residual needs to be corrected. As a result, though the model can have good predictive accuracy - we cannot use the `t-value` & `p-value` based analysis for statistical inference. Now under such circumstances, one can resort to a nonparametric bootstrap regression method for further statistical inference

**Residual Bootstrap Regression**
If the issue is only with distribution, then one can resort to residual bootstrap regression.

**Paired Bootstrap Regression**
 However, if homoscedasticity is violated, one can resort to paired Bootstrap regression.

**Possible code would look like**

```Julia
container=fit(@formula(y~x1+x2+x3),DataFrame,LinearRegression(),Bootstrap_Residual(),1000)
container=fit(@formula(y~x1+x2+x3),DataFrame,LinearRegression(),Bootstrap_Paired(),1000)
```
The `1000` is the Bootstrap simulation size

### PR DESCRIPTION
**Pull request by ShouvikGhosh2048**
Closes #100

### CODE REVIEW

File: src/frequentist/linear_regression.jl [hunk 1]
```diff
@@ -97,3 +97,92 @@ function fit(formula::FormulaTerm, data::DataFrame, modelClass::LinearRegression
...
+    ## This line is inefficient
+    A = vcov(model)/dispersion(model.model,true)
+
+    ## Once the Mousum's code is merged we will revert to following line
+    # A = GLM.inv(model)
+
+    β_star_matrix = zeros(sim_size,p)
+
+    for b in 1:sim_size
+        e_resample = rand(CRRao_rng, e, n)
```
Comment (ShouvikGhosh2048) [inline src/frequentist/linear_regression.jl:170]: I'm using `CRRao_rng` instead of `Random.seed`.

File: src/frequentist/linear_regression.jl [hunk 2]
```diff
@@ -97,3 +97,92 @@ function fit(formula::FormulaTerm, data::DataFrame, modelClass::LinearRegression
...
+    res = coeftable(model)
+    res = DataFrame(res)
+    e = residuals(model)
+
+    β_hat = coef(model)
+    p = length(β_hat)
+    n = size(X)[1]
+
+    ## This line is inefficient
+    A = vcov(model)/dispersion(model.model,true)
```
Comment (ShouvikGhosh2048) [inline src/frequentist/linear_regression.jl:162]: The example in the issue uses `model` but that errors out - `model.model` doesn't error - is this the correct value to pass?

File: src/frequentist/linear_regression.jl [hunk 3]
```diff
@@ -97,3 +97,92 @@ function fit(formula::FormulaTerm, data::DataFrame, modelClass::LinearRegression
...
+    y, X = modelcols(formula, data)
+
+    model = lm(formula, data)
+    res = coeftable(model)
+    res = DataFrame(res)
+    e = residuals(model)
+
+    β_hat = coef(model)
+    p = length(β_hat)
+    n = size(X)[1]
```
Comment (ShouvikGhosh2048) [inline src/frequentist/linear_regression.jl:159]: `nrow` in the example doesn't work as X is a matrix, not a DataFrame. I've used `size(X)[1]` instead.

File: src/frequentist/linear_regression.jl (hunk 1 above)
Comment (sourish-cmi) [reply src/frequentist/linear_regression.jl:170]: okay, that is a good idea.

File: src/frequentist/linear_regression.jl (hunk 2 above)
Comment (sourish-cmi) [reply src/frequentist/linear_regression.jl:162]: I believe this is correct. One test case to check is:

```Julia
B = inv(X'X)
```
The `B` should be exactly equal to `A`. However, I don't want to calculate the inverse for obvious reasons. If the GLM is already calculating it, then I would rather like to use it. My guess is `A` is throwing the correct answer.

Please check and let me know.

File: src/frequentist/linear_regression.jl (hunk 3 above)
Comment (sourish-cmi) [reply src/frequentist/linear_regression.jl:159]: Yes - that is correct.

**Review by sourish-cmi — APPROVED** [review]
changes are fine

**Pull request merged**